django/contrib/admin/views/main.py

```diff
@@ -114,9 +114,17 @@ def get_filters(self, request):
                 spec = list_filter(request, lookup_params, self.model, self.model_admin)
             else:
                 field_path = None
+                title = None
                 if isinstance(list_filter, (tuple, list)):
-                    # This is a custom FieldListFilter class for a given field.
-                    field, field_list_filter_class = list_filter
+                    field, second_param = list_filter
+
+                    if isinstance(second_param, str):
+                        # Second parameter is an override for the default title
+                        field_list_filter_class = FieldListFilter.create
+                        title = second_param
+                    else:
+                        # This is a custom FieldListFilter class for a given field.
+                        field_list_filter_class = second_param
```

Comment:
I'd write this block a little differently: 

```python
if isinstance(list_filter, (tuple, list)):
    # This is a custom FieldListFilter class for a given field.
    field, field_list_filter_class = list_filter
    if isinstance(field_list_filter_class, str):
        title = field_list_filter_class
        field_list_filter_class = FieldListFilter.create
else:
    ...
```

Reply:
I'm going to respectfully disagree here, but if you insist I'll rewrite it in this style.

IMO in your version the comment is incorrect (it's a custom `FieldListFilter` class **or** a `str`), and I think it's bad form (in general) for a variable to change types as it makes code trickier to reason about. In your code `field_list_filter_class` changes from `str` to a class, where as in my version `field_list_filter_class` is always a class. This also keeps the name of the variable accurate, so if in the future extra code gets inserted before the `isinstance` check no one has to scratch their head wondering why a variable labeling itself as a class is being checked to see if it's a `str` (especially since their initial thought may be that it's a dotted path to a class, but, nope...).

I will concede that `second_param` is not a great variable name. 😉